client/chat.coffee

```diff
@@ -122,6 +122,47 @@ nickEmail = (nick) ->
   n = model.Nicks.findOne canon: cn
   return model.getTag(n, 'Gravatar') or "#{cn}@#{settings.DEFAULT_HOST}"
 
+Template.starred_messages.onCreated ->
+  this.autorun =>
+    this.subscribe 'starred-messages', Session.get 'room_name'
+
+Template.starred_messages.helpers
+  messages: ->
+    # It would be nice to write a concatenation cursor, but it seems overkill
+    # for the number of starred messages we're ever likely to have.
+    r = []
+    for coll in [ model.OldMessages, model.Messages ]
+      c = coll.find {room_name: (Session.get 'room_name'), starred: true },
+        sort: [['timestamp', 'asc']]
+        transform: messageTransform
+      r.push c.fetch()...
+    r
+
+Template.media_message.events
+  'click .bb-message-star.starred': (event, template) ->
+    return unless $(event.target).closest('.can-modify-star').size() > 0
+    Meteor.call 'setStarred', this._id, false
+  'click .bb-message-star:not(.starred)': (event, template) ->
+    return unless $(event.target).closest('.can-modify-star').size() > 0
+    Meteor.call 'setStarred', this._id, true
+
+
+messageTransform = (m) ->
+  _id: m._id
+  message: m
+  # optional server-side follow-up info
+  followup: (( model.followupStyle() is 'server') and m.followup) or false
+  email: nickEmail m.nick
+  body: ->
```

Comment:
I know this looks weird, but it works. In fact I didn't notice I had forgot the 'do' until looking at it just now.